wallet-sdk/src/main/kotlin/org/stellar/walletsdk/exception/WalletException.kt

```diff
@@ -1,17 +1,27 @@
 package org.stellar.walletsdk.exception
 
 import okhttp3.Response
+import org.stellar.sdk.requests.ErrorResponse
+import org.stellar.walletsdk.util.toJsonOrNull
+
+data class AnchorErrorResponse(val error: String)
 
 sealed class WalletException : Exception {
   constructor(message: String) : super(message)
   constructor(message: String, cause: Exception) : super(message, cause)
 }
 
-// TODO: add arguments for error code, response message, etc.
-class NetworkRequestFailedException(
-  rawResponse: Response,
-  label: String? = "Request failed",
-) : WalletException("$label: $rawResponse")
+open class ServerRequestFailedException(response: Response) : WalletException("") {
```

Comment:
I think you should make response a val in case developer wants to access it (in both exceptions)